### PR TITLE
Tree search bar

### PR DESCRIPTION
**Pull request by jzbuchalski**
Update the search bar for the model & category trees.  The new search bar is responsive (using the iTwinUI ButtonGroup component) and we no longer support the "..." dropdown menu when the search box is displayed.  The request to remove the "..." came from the UX team (Gino). 

A boolean was added to show/hide the 2D & 3D tool buttons.  Gino requested we do not show these buttons by default.

image

image

image

### CODE REVIEW

File: packages/itwin/tree-widget/package.json [hunk 1]
```diff
@@ -59,6 +59,8 @@
     "@itwin/core-react": "^3.0.0",
     "@itwin/eslint-plugin": "^3.0.0",
     "@itwin/imodel-components-react": "^3.0.0",
+    "@itwin/itwinui-icons-react": "1.8.0",
```
Comment (aruniverse) [inline packages/itwin/tree-widget/package.json:62]: i dont think the icons need to be dev+peer and they can be a direct dep

File: packages/itwin/tree-widget/src/components/search-bar/SearchBar.tsx [hunk 2]
```diff
@@ -35,144 +50,196 @@ export interface SearchBarProps extends CommonProps {
...
-  private _searchBox = React.createRef<SearchBox>();
-
-  public static defaultProps: Partial<SearchBarProps> = {
-    alignment: Alignment.Left,
-    enableGrouping: true,
-  };
-
-  constructor(props: any) {
-    super(props);
+export const SearchBar = (props: SearchBarProps) => {
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/search-bar/SearchBar.tsx:64]: thanks for converting this to a fn component

File: packages/itwin/tree-widget/src/components/trees/ModelsTree.tsx [hunk 3]
```diff
@@ -24,8 +25,8 @@ interface TreeViewModelInfo {
 export const ModelsTreeComponent = (props: ModelTreeProps) => {
   const [is2dToggleActive, setIs2dToggleActive] = useState<boolean>(false);
   const [is3dToggleActive, setIs3dToggleActive] = useState<boolean>(false);
-  const [icon2dToggle, setIcon2dToggle] = useState<string>("icon-visibility");
-  const [icon3dToggle, setIcon3dToggle] = useState<string>("icon-visibility");
+  const [icon2dToggle, setIcon2dToggle] = useState(<SvgVisibilityShow />);
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/trees/ModelsTree.tsx:28]: this makes sense, but does this seem weird to you? storing a component in state?

Comment (JoeZman) [reply packages/itwin/tree-widget/src/components/trees/ModelsTree.tsx:28]: Let me take a look, not sure why this was necessary to be in state, can't imagine any caller setting those icons.

Comment (JoeZman) [reply packages/itwin/tree-widget/src/components/trees/ModelsTree.tsx]: Removed from state.

File: packages/itwin/tree-widget/package.json (hunk 1 above)
Comment (JoeZman) [reply packages/itwin/tree-widget/package.json]: Done

**Review by aruniverse — APPROVED** [review]
Did a quick pass lgtm, maybe get another approval from @jason-crow, up to you.

File: packages/itwin/tree-widget/package.json [hunk 4]
```diff
@@ -81,6 +83,8 @@
     "@itwin/components-react": "^3.0.0",
     "@itwin/core-frontend": "^3.0.0",
     "@itwin/core-react": "^3.0.0",
+    "@itwin/itwinui-icons-react": "^1.8.0",
```
Comment (aruniverse) [inline packages/itwin/tree-widget/package.json:86]: the icon pkgs should not be peer deps, they should just be direct deps
i also dont agree with itwinui-react be listed as dev+peer

**Comment by grigasp** [comment]
@aruniverse @jzbuchalski should this be closed?

**Pull request closed without merging**